### PR TITLE
ETARGET Bid adapter: add get meta data functionality

### PR DESCRIPTION
**Pull request by etargetse**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@etargetse looks like you have some linting errors on this pr

**Comment by patmmccann** [comment]
tagging as taxonomy, likely needs committeee discussion

**Comment by etargetse** [comment]
@ChrisHuie can you please check what is wrong with etargetBidAdapter.md file? circleci faild test but details reveals nothing, just infinite loading window so i have no info whats wrong. Thanks.

**Comment by ChrisHuie** [comment]
Here are the linting errors I am seeing in CircleCi. Not seeing anything wrong with the md file? We don't really check markdown files with tests just code files.

/home/circleci/Prebid.js/modules/etargetBidAdapter.js
  27:26  error  Missing space before opening brace                           space-before-blocks
  30:26  error  A space is required after ','                                comma-spacing
  30:37  error  A space is required after ','                                comma-spacing
  30:51  error  A space is required after ','                                comma-spacing
  30:68  error  A space is required after ','                                comma-spacing
  30:77  error  A space is required after ','                                comma-spacing
  30:84  error  A space is required after ','                                comma-spacing
  31:5   error  Expected space(s) after "try"                                keyword-spacing
  32:7   error  Expected space(s) after "for"                                keyword-spacing
  32:27  error  Missing space before opening brace                           space-before-blocks
  33:1   error  Expected indentation of 8 spaces but found 10                indent
  33:11  error  Expected space(s) after "if"                                 keyword-spacing
  33:14  error  Unary word operator 'typeof' must be followed by whitespace  space-unary-ops
  33:31  error  Infix operators must be spaced                               space-infix-ops
  33:42  error  Missing space before opening brace                           space-before-blocks
  34:1   error  Expected indentation of 10 spaces but found 14               indent
  35:1   error  Expected indentation of 10 spaces but found 14               indent
  36:1   error  Expected indentation of 10 spaces but found 14               indent
  36:15  error  Expected space(s) after "if"                                 keyword-spacing
  36:34  error  Infix operators must be spaced                               space-infix-ops
  36:36  error  Strings must use singlequote                                 quotes
  36:54  error  Missing space before opening brace                           space-before-blocks
  37:1   error  Expected indentation of 12 spaces but found 18               indent
  37:19  error  Expected space(s) after "if"                                 keyword-spacing
  37:43  error  Infix operators must be spaced                               space-infix-ops
  37:47  error  Missing space before opening brace                           space-before-blocks
  38:1   error  Expected indentation of 14 spaces but found 22               indent
  38:23  error  Expected space(s) after "if"                                 keyword-spacing
  38:38  error  Missing space before opening brace                           space-before-blocks
  39:1   error  Expected indentation of 16 spaces but found 26               indent
  40:1   error  Expected indentation of 14 spaces but found 22               indent
  40:24  error  Trailing spaces not allowed                                  no-trailing-spaces
  41:1   error  Expected indentation of 14 spaces but found 22               indent
  42:1   error  Expected indentation of 12 spaces but found 18               indent
  43:1   error  Expected indentation of 10 spaces but found 14               indent
  44:1   error  Expected indentation of 8 spaces but found 10                indent
  49:6   error  Expected space(s) before "catch"                             keyword-spacing
  49:6   error  Expected space(s) after "catch"                              keyword-spacing
  49:14  error  Missing space before opening brace                           space-before-blocks
  87:17  error  'getMetaData' is not defined                                 no-undef

/home/circleci/Prebid.js/test/spec/modules/etargetBidAdapter_spec.js
  21:1  error  Trailing spaces not allowed  no-trailing-spaces

✖ 41 problems (41 errors, 0 warnings)
  40 errors, 0 warnings potentially fixable with the `--fix` option.

**Comment by etargetse** [comment]
@ChrisHuie can you explain for me error which are now shown in circleci ? Thanks .

**Comment by ChrisHuie** [comment]
@etargetse it looks like CircleCi is currently experiencing an outage so tests are hanging up at the moment.

**Comment by etargetse** [comment]
@ChrisHuie circleci still fails on Firefox 73, can you explain why? from the log it is not clear. Thanks.

**Comment by ChrisHuie** [comment]
@etargetse I'm not 100% sure but think it is either unsupported strings or data types are being passed on the end-to-end tests or there is latency being introduced somewhere that is impacting multiple types of adapters

**Comment by etargetse** [comment]
so when i dont know how to fix problem or event if there is problem, what should i do next? can we pass test manualy without circleci?

File: modules/etargetBidAdapter.js [hunk 1]
```diff
@@ -57,18 +58,45 @@ export const spec = {
...
       netRevenue: netRevenue,
+      metaData: getMetaData(),
       bidder: 'etarget',
       gdpr: gdprObject
     };
 
+    function getMetaData() {
+      var mts = {};
+      var hmetas = document.getElementsByTagName('meta');
+      var wnames = ['title', 'og:title', 'description', 'og:description', 'og:url', 'base', 'keywords'];
```
Comment (patmmccann) [inline modules/etargetBidAdapter.js:69]: can you look in the ortb2 object for this stuff as well?

Publishers or a prebid module might be setting it directly 
https://github.com/prebid/Prebid.js/blob/585c7107645ee9f79befb9d17d2cf8c6000b0d00/modules/enrichmentFpdModule.js#L64

or via methods in #5795

Comment (etargetse) [reply modules/etargetBidAdapter.js:69]: Thanks for recommendation, but this is not solving my problem. With or without my new function "getMetaData" circleci is failing tests.

Comment (patmmccann) [reply modules/etargetBidAdapter.js:69]: I'm aware, this is something else you should fix

**Comment by ChrisHuie** [comment]
@etargetse I messed around with this a bit and think pulling in upstream commits from master might automatically fix the issue. This pr is a bit behind the main branch.

**Comment by etargetse** [comment]
@ChrisHuie ok i will close this pull request and try new one on master, thanks

**Pull request closed without merging**